### ISSUE TITLE
Random freezing.

### ISSUE THREAD

**Issue by Giger22** [mullvad/mullvad-browser#203]
When opening link or any other action Mullvad browser randomly freezes for a few seconds, especially after a hibernation.

**Comment by Giger22** [comment]
Issue is also activated by opening a link from a different application (i.e. Freetube).

**Comment by Giger22** [comment]
I found that issue is resolved if you restart plasma (kstart plasmashell).

**Comment by Giger22** [comment]
Ok, I checked once more and this time resetting a plasmashell did not help with the issue.

**Comment by Giger22** [comment]
Issue persist with 13.0.9.

**Comment by Giger22** [comment]
Same with 13.0.10. I noticed that sometimes process called RDD takes up to 13% when mullvad is freezed.

**Comment by Giger22** [comment]
13.0.12 slightly improved freeze time.

**Comment by Giger22** [comment]
I noticed that logging out and then in solves this. Issue persis with KDE Plasam 6.0.1.

**Comment by ruihildt** [comment]
Do you have that only with Mullvad Browser or also with other Firefox-based browsers?

**Comment by Giger22** [comment]
Only with Mullvad browser and I regularly use both Firefox, Librewolf and Tor.

**Comment by Giger22** [comment]
It persist with 13.0.14.

**Comment by vladyslavfox** [comment]
I think I have the same problem. It's usually happening when using some heavy websites, like Telegram Web or Unsplash. What I also noticed is that the CPU usage increases during the freeze and the laptop starts to turn on the fan.

**Comment by Giger22** [comment]
> I think I have the same problem. It's usually happening when using some heavy websites, like Telegram Web or Unsplash. What I also noticed is that the CPU usage increases during the freeze and the laptop starts to turn on the fan.
> 
> I'm running Debian Bookworm. And the problem only happens with the Mullvad browser. Other FF-based browsers - like LibreWolf, Tor and FF - work fine.

In my case it is completely random.

**Comment by vladyslavfox** [comment]
> In my case it is completely random.

Maybe it's another bug, maybe not. But some sites freeze for me more often than others. 

After several days of use of the Mullvad browser on an M1 Macbook, I can add that the problem is there too, so it is not platform-specific.

**Comment by ruihildt** [comment]
@Giger22 @vladyslavfox Can you try downloading the [Mullvad Browser Alpha 14.0a4](https://github.com/mullvad/mullvad-browser/releases/tag/14.0a4) and see if you still have random freezes?

Thanks!

**Comment by Giger22** [comment]
Nope, I have experienced the same freezing when testing alpha. What is weird the freezing appears to occur at the same time... like some other underlying process causing it.

**Comment by ruihildt** [comment]
> What is weird the freezing appears to occur at the same time... like some other underlying process causing it.

What do you mean by "at the same time"?

**Comment by Giger22** [comment]
> > What is weird the freezing appears to occur at the same time... like some other underlying process causing it.
> 
> What do you mean by "at the same time"?

When I encountered a freezing using the stable release I switched over to the alpha version. Both browsers were experiencing freezing at the same time.

**Comment by ruihildt** [comment]
In order to further debug this issue, you would need to be comfortable using the Firefox Profiler (https://profiler.firefox.com/docs/#/).

Is that something that you would be open to do?

**Comment by Giger22** [comment]
Yes, I am open to do it. What to look for?

**Comment by Giger22** [comment]
[Here](https://share.firefox.dev/3Yl76Ua) is data when I was experiencing the freezing. Only at the end (last ~10 seconds) the freezing ended and I could use the browser without it. I downloaded data with hidden threads and time ranges, if needed I could DM you.

**Comment by Giger22** [comment]
I found something strange. This bug sometimes occurs when I clear my clipboard, and when I copy any letter or word, the freezing stops. The moment the freezing happens is definitely during the "Jank - event processing delay."

**Comment by hackademix** [comment]
> This bug sometimes occurs when my clipboard is empty, but when I copy any letter or word, the freezing stops.

Thanks. 

May I assume this means "when you copy something from an application different than the browser and then you switch back to the browser window"?

**Comment by Giger22** [comment]
No, It does not matter where I copy from. Sometimes the stalling stops when I just copy a letter (I tested the browser and KWrite).

**Comment by ruihildt** [comment]
@Giger22 Can you try the [latest alpha](https://github.com/mullvad/mullvad-browser/releases/tag/14.0a10)?

There's a potential fix included for your issue.

**Comment by Giger22** [comment]
I have noticed that the stalling time is shorter but this still happens.

**Comment by Giger22** [comment]
It still happens. In my case only when my clipboard have cleared history.

**Comment by ruihildt** [comment]
@Giger22 Can you reproduce this issue when you use the standalone install (using the tarball)?

**Comment by Giger22** [comment]
Yes, both browsers have the same behavior. Once I clear my clipboard the freezing starts, the same moment I copy even one letter to my clipboard it stops.

**Comment by ClaudiusGoennus** [comment]
It's also happening for me. Im on Artix Linux, Kernel 6.12.73-1-lts, using i3 Window Manager. It's only happening in Mullvad Browser, not on Firefox. It's also accompanied by a Spike in one or two CPU Cores and happens regardless of what site im browsing and also when im just using the Browser UI. Copying stuff to my clipboard doesn't change anything.

[This is a Profiler capture](https://profiler.firefox.com/public/6zytf9b3e6z6cx1d99kyjntvtq9ax1yk0tnyrhr/calltree/?globalTrackOrder=01&thread=1&v=15), I just clicked back and forth between this thread and the Firefox Profiler website, Freezes occured right at the beginning and around 11.1 Seconds.
